pages/doc-components/CodeContainer.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import {LiveCode as MDXLive} from 'mdx-live'
+import copy from 'clipboard-copy'
+
+import {Relative, Absolute, Link, StyledOcticon} from '../..'
+import {Clippy, Check} from '@githubprimer/octicons-react'
+
+function CodeContainer({children: code, className: language}) {
+  if (language === 'language-.jsx') return <LiveCode initialCode={code} />
+  else return <StaticCode code={code} />
+}
+
+function LiveCode({initialCode}) {
+  const [code, setCode] = React.useState(initialCode)
+
+  return (
+    <Relative>
+      <MDXLive code={code} editorProps={{onChange: code => setCode(code)}} />
+      <CopyButton code={code} />
+    </Relative>
+  )
+}
+
+function StaticCode({code}) {
+  return (
+    <Relative>
+      <pre>{code}</pre>
+      <CopyButton code={code} />
+    </Relative>
+  )
+}
+
+/* Totally magic number based on what feels-right™ */
+const COPY_RESET_TIME = 1500
```

Comment:
Reset to copy button after 1.5 seconds for more changes

![reset](https://user-images.githubusercontent.com/1863771/58264319-486cbf00-7d9b-11e9-9d12-9b4463a8374f.gif)